### PR TITLE
GH-3509: fix support metric

### PR DESCRIPTION
**Pull request by MattGPT-ai**
Addresses https://github.com/flairNLP/flair/issues/3509

Also adds more type hinting and fixes mutable default arguments.

### CODE REVIEW

**Comment by MattGPT-ai** [comment]
I'm not sure what the error occurring is, it doesn't seem to be related to my changes but maybe you are familiar with it

**Comment by MattGPT-ai** [comment]
Looks like the test passed without me changing anything, so this should be ready for review

File: flair/nn/model.py [hunk 1]
```diff
@@ -737,7 +743,8 @@ def _encode_data_points(self, sentences: List[DT], data_points: List[DT2]):
 
         return data_point_tensor
 
-    def _mask_scores(self, scores, data_points):
+    def _mask_scores(self, scores: Tensor, data_points) -> Tensor:
+        """This appears to have not been fully implemented and simply returns scores."""
```
Comment (alanakbik) [inline flair/nn/model.py:747]: In most models, scores are not masked, so the default implementation is to just return the scores. The `SpanClassifier` overwrites this method to account for "candidate lists". Anything not on these lists, will be masked.

Comment (alanakbik) [inline flair/nn/model.py:747]:
```suggestion
        """Classes that inherit from DefaultClassifier may optionally mask scores."""
```

**Review by alanakbik — CHANGES_REQUESTED** [review]
Looks good, thanks for adding this! Just needs a tiny modification in one of the comment lines, as otherwise it would be misleading.

**Comment by alanakbik** [comment]
@MattGPT-ai thanks for fixing this!

**Pull request merged**